src/contrib/inference/dynamichmc.jl

```diff
@@ -55,44 +64,41 @@ function DynamicPPL.initialstep(
     model::Model,
     spl::Sampler{<:DynamicNUTS},
     vi::AbstractVarInfo;
-    N::Int,
     kwargs...
 )
-    # Set up lp function.
-    function _lp(x)
-        gradient_logp(x, vi, model, spl)
-    end
-
-    link!(vi, spl)
-    l, dl = _lp(vi[spl])
-    while !isfinite(l) || !isfinite(dl)
-        model(vi, SampleFromUniform())
-        link!(vi, spl)
-        l, dl = _lp(vi[spl])
-    end
-
-    if spl.selector.tag == :default && !islinked(vi, spl)
-        link!(vi, spl)
-        model(vi, spl)
+    # Ensure that initial sample is in unconstrained space.
+    if !DynamicPPL.islinked(vi, spl)
+        @debug "X -> R..."
```

Comment:
I feel like we should maybe add this to the definition of `link!` so that we don't have random idiosyncratic debug calls for some samplers and not others.

Reply:
https://github.com/TuringLang/DynamicPPL.jl/pull/206#issue-544639845